### PR TITLE
[DF-35] User register with validation data

### PR DESCRIPTION
**Pull request by pcybxine**
endpoints
- ~~POST: /user/register to register~~
- ~~GET: /user?skip=0&limit=10 to list all users~~

deprecated all controllers

resolvers
- user.resolver

schema
- user.schema.ts
- user-token.schema.ts

others
- ~~MongoDB Exception utils in `common/exception/mongo-error.exception.ts`~~
- Add `EmailDuplicatedException` that catch from `IndexDuplicated`

### CODE REVIEW

File: test/user.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      imports: [UserModule],
+    })
+      .overrideProvider(DATABASE_CONNECTION)
+      .useFactory({
+        factory: async () => await mockDatabaseFactory(),
+      })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe());
```
Comment (yee2542) [inline test/user.e2e-spec.ts:34]: set to use `transform:true`

File: src/user/pipes/user.trim.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import {
+  ArgumentMetadata,
+  BadRequestException,
+  Injectable,
+  PipeTransform,
+} from '@nestjs/common';
+
+const EXCEPT_FIELD = ['password'];
+
```
Comment (yee2542) [inline src/user/pipes/user.trim.pipe.ts:9]: may be we add `INCLUEDED_FIELD` to select only trim field to preventing for loss performance

File: src/database/database.provider.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export const databaseProviders: Provider[] = [
+  {
+    provide: DATABASE_CONNECTION,
+    useFactory: async (): Promise<typeof mongoose> => {
+      return mongoose.connect(
+        'mongodb://mongodb-sharded:27017/fluent-search-bff',
+        {
+          user: 'root',
+          pass: 'fluent-search-bff-db',
```
Comment (yee2542) [inline src/database/database.provider.ts:13]: use this value from `ConfigModule`

**Review by yee2542 — CHANGES_REQUESTED** [review]
Check

File: test/user-resolver.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    }
+  }
+  `;
+    return request(app.getHttpServer())
+      .post('/graphql')
+      .send({
+        query: userInput,
+      })
+      .expect(res => {
+        console.log(res.body.data);
```
Comment (octobrekids) [inline test/user-resolver.e2e-spec.ts]: Don't forget to remove after debug ka

File: src/user/user.module.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Module } from '@nestjs/common';
+import { ConfigModule } from '../config/config.module';
+import { DatabaseModule } from '../database/database.module';
+import { userProivders } from './user.providers';
+import { UserResolver } from './user.resolver';
+import { UserService } from './user.service';
+
+@Module({
+  imports: [DatabaseModule, ConfigModule],
+  providers: [UserService, UserResolver, ...userProivders],
```
Comment (octobrekids) [inline src/user/user.module.ts:10]: please check spelling of `userProviders`

File: src/user/user.providers.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { Provider } from '@nestjs/common';
+import { Connection } from 'mongoose';
+import { DATABASE_CONNECTION } from '../database/constants/database.constant';
+import { USER_COLLECTION } from './constants/user.collection.constant';
+import { USER_MODEL } from './constants/user.provider.constant';
+import userSchema from './schemas/user.schema';
+
+export const userProivders: Provider[] = [
```
Comment (octobrekids) [inline src/user/user.providers.ts:8]: please check `userProviders` spelling

**Review by octobrekids — COMMENTED** [review]
Please check ka

**Review by yee2542 — APPROVED** [review]

**Pull request merged**